Install sequencing report service

Aratz opened the pull request:
Install the sequencing report service on Miarka

This is mostly copied from the `arteria-delivery` role since both services are quite similar.

There is one line I'm not sure about
https://github.com/Aratz/miarka-provision/blob/40923458259d48918f08cf7d35abb54f3203859e/roles/sequencing-report-ws/tasks/main.yml#L9

b97pla commented on roles/sequencing-report-ws/tasks/main.yml:
```diff
@@ -0,0 +1,15 @@
+
+  - name: add static folders to be created
+    set_fact:
+      static_folders: "{{ static_folders + [arteria_service_log_dir, runfolder_path, project_path, staging_path, delivery_links] }}" # TODO check if we need this
```
don't think you need all of these, right? should only need the dirs the service is actually reading/writing to/from

Aratz replied:
Yeah I wasn't quite sure what these locations were for, I'm guessing we only need `arteria_service_log_dir` and `runfolder_path` for the sequencing report service? Can you confirm this @matrulda?

b97pla commented on roles/sequencing-report-ws/defaults/main.yml:
```diff
@@ -0,0 +1,27 @@
+# staging sync.
+
+seqreports_nf_profile: singularity,uppmax
+fastq_ref_genomes_path: $FASTQ_SCREEN_DATA
```
ah ok, saw the module system call below 👇

Aratz replied:
Yes, I've been discussing this with Uppmax, since igenomes didn't quite match our needs, they've installed the fastq reference genomes for us instead.

b97pla commented on roles/setup_base_config/templates/sourceme_site.sh.j2:
```diff
@@ -14,8 +17,8 @@ alias deployed_env="cat {{ deployed_tool_versions }}"
 
 {% if site == "upps" %}
 export PATH={{ sw_path }}:$PATH
+module load bioinfo-tools bowtie2 fastq_screen
```
in keeping with the `DDS-cli` pattern in the delivery service, would it be possible to only load these modules for the seqreports service through a wrapper? or is it required that the user also has these modules loaded in any way?

Aratz replied:
Yes I think that should be possible, I'll look into that :+1:

matrulda replied:
Note that seqreports uses a singularity container to run fastq_screen, so the module won't be used. 

If we only load the module to get the path to the references we could perhaps directly specify it ? `/sw/data/fastq_screen_data/20220330/miarka/FastQ_Screen_Genomes`

b97pla reviewed (approved):
nice work 👍 

looks good to me but a few comments/questions. I think testing this ahead of deployment will probably be the most important check

matrulda commented on roles/sequencing-report-ws/defaults/main.yml:
```diff
@@ -0,0 +1,27 @@
+# staging sync.
+
+seqreports_nf_profile: singularity,uppmax
+fastq_ref_genomes_path: $FASTQ_SCREEN_DATA
```
Ah realized now that it is when the module is loaded.

matrulda reviewed (commented):
Looks good! Left some comments and a general one that I'll add here: 
In our current setup the seqreports pipeline will download images it needs on the fly and put them in `$NXF_WORK/singularity`. This does not play well with Miarka being a closed node (Not with our current setup either, but we have implemented a workaround https://snpseq.atlassian.net/browse/DEVELOP-1600). We need to add a step where images are downloaded. It could perhaps be done by running the integraton test of the pipeline. Alternatively we could parse this file and download the images.

Aratz commented:
Thanks a lot for the feedback! I think I've addressed most points now.

One small comment is that seqreport's `uppmax` profile is still called `irma`, this is fixed on `seqreports/dev` but it has not been released yet.

matrulda commented on roles/arteria-sequencing-report-ws/tasks/install.yml:
```diff
@@ -0,0 +1,36 @@
+  args:
+    chdir: "{{ arteria_service_src_path }}/seqreports"
+  environment:
+    NXF_SINGULARITY_CACHEDIR: "{{ nf_core_vars.NXF_SINGULARITY_CACHEDIR }}"
```
Is it possible to access this variable without `https://docs.ansible.com/ansible/latest/collections/ansible/builtin/include_vars_module.html` or adding it as a depenency role?

Aratz replied:
Oh yes good catch! I'll put the dependency back :+1:

matrulda commented on roles/arteria-sequencing-report-ws/tasks/install_config.yml:
```diff
@@ -0,0 +1,42 @@
+    dest: "{{ arteria_service_config_root }}/logger.config"
+
+- name: deploying {{ arteria_service_name }} checkqc config
+  template:
```
Since we don't substitute anything in the template we could add it to files instead and use `copy`.

matrulda commented on roles/sequencing-report-ws/tasks/install_config.yml:
```diff
@@ -0,0 +1,36 @@
+    src: seqreport_logger.config.j2
+    dest: "{{ arteria_service_config_root }}/logger.config"
+
+- name: deploying {{ arteria_service_name }} checkqc config
```
@Aratz Do you think it would be OK to add a line to the config specifying which version it is taken from? I think this will help us.

Aratz replied:
Yes, good idea, will do :+1:

matrulda reviewed (commented):
Look good! :+1:  Left some comments you can have a look at.

Aratz commented:
Great thanks!

matrulda commented on roles/arteria-sequencing-report-ws/tasks/install.yml:
```diff
@@ -0,0 +1,36 @@
+    module load bioinfo-tools Nextflow/21.04
+    nextflow run main.nf -profile test,singularity
+  args:
+    chdir: "{{ arteria_service_src_path }}/seqreports"
```
Sorry for not commenting about this before, but I realized this now: It is probably best to run this on scratch or some temp folder to avoid results being written to this folder.

matrulda replied:
Note: it is not only the results folder that will be generated in that folder, also the `.nextflow` folder + `.nextflow.log`.

matrulda reviewed (commented):
Changes look good! :ok But I realized another thing we might wanna change (sorry)

matrulda reviewed (changes requested):
Think I found a typo @Aratz

Aratz commented on roles/arteria-sequencing-report-ws/tasks/install.yml:
```diff
@@ -27,10 +27,11 @@
+  shell: |
     module load bioinfo-tools Nextflow/21.04
-    nextflow run main.nf -profile test,singularity
+    nextflow -log /scratch run {{ arteria_service_src_path }}/seqreportsmain.nf -profile test,singularity
```
Yes, good catch!

Aratz commented on roles/arteria-sequencing-report-ws/tasks/install.yml:
```diff
@@ -27,10 +27,11 @@
+    nextflow -log /scratch run {{ arteria_service_src_path }}/seqreportsmain.nf -profile test,singularity
   args:
-    chdir: "{{ arteria_service_src_path }}/seqreports"
+    chdir: /scratch
```
Done!

matrulda commented on roles/arteria-sequencing-report-ws/tasks/install.yml:
```diff
@@ -29,9 +29,9 @@
   shell: |
     module load bioinfo-tools Nextflow/21.04
-    nextflow -log /scratch run {{ arteria_service_src_path }}/seqreportsmain.nf -profile test,singularity
+    nextflow -log /scratch run {{ arteria_service_src_path }}/seqreports/main.nf -profile test,singularity
```
Could you replace scratch with the ENV parameter here as well?

Aratz replied:
Done!

matrulda reviewed (approved):
Excellent, great work :+1:
